### PR TITLE
Adding Recipe Result back in

### PR DESCRIPTION
**Pull request by Jetski5822**
Using IRecipeResultAccessor, and also an index - I was thinking of adding a IRecipeStepResultAccessor, but as the RecipeResult is one document with many steps, I think it should be okay? Let me know

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.Recipes/Services/RecipeStepExecutor.cs [hunk 1]
```diff
@@ -60,24 +63,18 @@ public class RecipeStepExecutor : IRecipeStepExecutor
             bool IsSuccessful,
             Exception exception = null)
         {
-            // TODO: bad pattern, use an index
+            var recipeResult = _recipeResultAccessor
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Recipes/Services/RecipeStepExecutor.cs]: It's an async method, await this call.

**Comment by sebastienros** [comment]
Ok to have a single document. Then because recipes are run in one step, why not persist it only in success or failure with all its steps. So the `GetRecipe(executionId)` is looking up for in memory objects (a dictionary of recipes) which means it's currently runnging, and fallbacks to the database if it's terminated (success or failure).

File: src/Orchard.Web/Modules/Orchard.Recipes/Services/RecipeStepExecutor.cs (hunk 1 above)
Comment (Jetski5822) [inline src/Orchard.Web/Modules/Orchard.Recipes/Services/RecipeStepExecutor.cs]: Ok

**Comment by Jetski5822** [comment]
good point, thought about that, and will change

**Comment by sebastienros** [comment]
Because of the merge issues, I applied it manually

**Pull request closed without merging**